draft-barnes-acme.md

```diff
@@ -1140,45 +1140,96 @@ If the server is able to validate the signature and confirm that the jwk and alg
 }
 
 ~~~~~~~~~~
 
-In response to this challenge, the client first MUST verify that the token contains only ASCII characters.  If so, the client constructs the validation domain name by appending the label "_acme-challenge" to the domain name being validated.  For example, if the domain name being validated is "example.com", then the client would provision the following DNS record:
+In response to this challenge, the client computes the value it will provision to the DNS as the MAC over a JSON object.  The validation object covered by the MAC has the following fields:
+
+type (required, string):
+: The string "dns"
+
+token (required, string):
+: The token value in the challenge
 
+accountKey (required, JWK):
+: The client's account public key
+
+~~~~~~~~~~
+{
+  "type": "dns",
+  "token": "17817c66b60ce2e4012dfad92657527a",
+  "accountKey": { "kty": "RSA", ... }
+}
 ~~~~~~~~~~
 
-_acme-challenge.example.com. IN TXT "17817c66b60ce2e4012dfad92657527a"
+The to compute the MAC key, the client generates an ECDH key pair and uses it with the serverPublicKey value in the challenge to generate a 256-bit shared secret according to the algorithm defined in {{key-agreement}}.
+
+Once the client has generated the validation object and the MAC key, it serializes the validation object to UTF-8, and forms an HMAC-based JWS with the JSON object as its payload.  This JWS is NOT REQUIRED to have the "nonce" header parameter.
+
+The record provisioned to the DNS is the "signature" value from the JWS, i.e., the base64-encoded MAC value.  If so, the client constructs the validation domain name by appending the label "_acme-challenge" to the domain name being validated, then provisions a TXT record with the signature value under that name. For example, if the domain name being validated is "example.com", then the client would provision the following DNS record:
+
+~~~~~~~~~~
+
+_acme-challenge.example.com. IN TXT "gfj9XqFv07e1wU66hSLYkiFqYakPSjAu8TsyXRg85nM"
 
 ~~~~~~~~~~
 
-The response to a DNS challenge is simply an acknowledgement that the relevant record has been provisioned.
+The response to a DNS challenge provides the ECDH key that the client generated and the HMAC-based JWS object.
 
 type (required, string):
 : The string "dns"
 
+clientPublicKey (required, JWK):
+: The ECDH key generated by the client
+
+validation (required, JWS):
+: The JWS object computed with the validation object and the derived key
+
 ~~~~~~~~~~
 
 {
   "type": "dns"
+  "clientPublicKey": { "kty": "EC", ... },
+  "validation": {
+    "header": { "alg": "HS256" },
+    "payload": "qzu9...6bjn",
+    "signature": "r32O...ge8Y"
+  }
 }
 
 ~~~~~~~~~~
 
-To validate a DNS challenge, the server queries for TXT records under the validation domain name.  If it receives a record whose contents match the token in the challenge, then the validation succeeds.  Otherwise, the validation fails.
+To validate a DNS challenge, the server performs the following steps:
```

Comment:
Have it your way.  Six of one / half dozen of the other -- it just depends on whether you think an ECDH computation or a DNS query is more expensive.